File: c++/Strictly Palindromic Number.cpp
#include <string>
#include <vector>
#include <cassert>

class Solution {
public:
    bool isStrictlyPalindromic(int n) {
        if (n <= 2) return true;
        for (int base = 2; base < n - 1; ++base) {
            if (!isPalindromic(n, base)) return false;
        }
        return true;
    }
private:
    bool isPalindromic(int n, int base) {
        std::string numInBase = convertToBase(n, base);
        int left = 0, right = numInBase.size() - 1;
        while (left < right) {
            if (numInBase[left++] != numInBase[right--]) return false;
        }
        return true;
    }
    std::string convertToBase(int n, int base) {
        std::string result = "";
        while (n > 0) {
            result = std::to_string(n % base) + result;
            n /= base;
        }
        return result;
    }
};

// test case
int main() {
    Solution sol;
    
    assert(sol.isStrictlyPalindromic(6) == true);
    
    assert(sol.isStrictlyPalindromic(11) == false);
    
    return 0;
}

File: python/Strictly Palindromic Number.py
class Solution:
    def isStrictlyPalindromic(self, n: int) -> bool:
        if n <= 2:
            return True
        for base in range(2, n - 1):
            if not self.isPalindromic(n, base):
                return False
        return True

    def isPalindromic(self, n: int, base: int) -> bool:
        num_in_base = self.convertToBase(n, base)
        left, right = 0, len(num_in_base) - 1
        while left < right:
            if num_in_base[left] != num_in_base[right]:
                return False
            left += 1
            right -= 1
        return True

    def convertToBase(self, n: int, base: int) -> str:
        result = ""
        while n > 0:
            result = str(n % base) + result
            n //= base
        return result